packages/buidler-core/src/internal/core/providers/construction.ts

```diff
@@ -52,7 +52,7 @@ export function createProvider(
     provider = new HttpProvider(
       httpNetConfig.url!,
       networkName,
-      undefined,
+      httpNetConfig.headers,
```

Comment:
I can't believe this was already there and not being exposed to the users 😅

Reply:
Really happy to know there's equal interest to include it <3